cms/djangoapps/contentstore/views/component.py

```diff
@@ -339,16 +340,21 @@ def create_support_legend_dict():
                             tab = 'advanced'
                         hinted = template.get('hinted', False)
 
-                        templates_for_category.append(
-                            create_template_dict(
-                                _(template['metadata'].get('display_name')),  # lint-amnesty, pylint: disable=translation-of-non-string
-                                category,
-                                support_level_with_template,
-                                template_id,
-                                tab,
-                                hinted,
+                        if use_new_text_editor() and template_id == 'raw.yaml':
```

Comment:
Can you explain what these lines of code do? They aren't explanatory. Why are we not making a dict for those two cases?

Reply:
These two lines are checking if the new text editor flag is on and skipping one or the other `.yaml` files depending on the flag value. At the moment to change the default text for the Raw HTML editor, there needs to be a corresponding `.yaml` file for a specific button. The buttons are created in the wrapping for loop. But I couldn't find a way to filter the `.yaml` files entering the loop so I used the continue as a filter.

Reply:
Oh ok, so essentially it is "don't add two templates for raw, just add one." Makes sense.